tests/integration/tests/test_sanity_telemetry.py

```diff
@@ -0,0 +1,123 @@
+"""
+This test suite goal is to validate end-to-end cloud security telemetry payloads.
+The suite assumes that the environment is already deployed and all integrations,
+such as KSPM (Kubernetes Security Posture Management) and CSPM (Cloud Security Posture Management),
+have been created.
+The goal of this suite is to perform basic sanity checks to verify the telemetry fetchers are working
+ as expected.
+"""
+# pylint: disable=redefined-outer-name
+import pytest
+from commonlib.utils import get_telemetry
+from configuration import elasticsearch
+from munch import munchify
+
+
+@pytest.fixture(scope="module")
+def cloud_security_telemetry_data():
+    """Fixture to fetch telemetry data"""
+    telemetry_payload = get_telemetry(elasticsearch)
+    telemetry_object = munchify(telemetry_payload[0])
+    return telemetry_object.stats.stack_stats.kibana.plugins.cloud_security_posture
+
+
+@pytest.mark.sanity
+def test_telemetry_indices(cloud_security_telemetry_data):
+    """
+    Test case to check telemetry indices are fetched as expected
+
+    Raises:
+        AssertionError: If one of the payload keys is missing.
+    """
+    indices_stats = cloud_security_telemetry_data.indices
+    indices = [
+        "findings",
+        "latest_findings",
+        "vulnerabilities",
+        "latest_vulnerabilities",
+        "score",
+    ]
+
+    # indices stats
+    for index in indices:
+        assert indices_stats[index].doc_count > 0, f"Expected {index} index to contain data"
+
+
+@pytest.mark.sanity
+def test_telemetry_cloud_account_stats(cloud_security_telemetry_data):
+    """
+    Test case to check telemetry cloud account stats are fetched as expected
+
+    Raises:
+        AssertionError: If one of the payload keys is missing.
+    """
+    # account stats
+    cloud_account_stats = cloud_security_telemetry_data.cloud_account_stats
+    for account in cloud_account_stats:
+        assert (
+            len(account.account_id) > 0
+        ), f"Telemetry data missing account_id for \
+            cloud_account_stats {account}"
+
+        assert (
+            len(account.product) > 0
+        ), f"Telemetry data missing product for \
+             cloud_account_stats {account}"
+
+        if not (account.product == "kspm" and "CIS Kubernetes" in account.posture_management_stats.benchmark_name):
+            assert (
+                len(account.cloud_provider) > 0
+            ), f"Telemetry data missing cloud_provider for \
+                    cloud_account_stats {account}"
+        if account.product != "vuln_mgmt":
+            assert (
+                len(account.posture_management_stats.benchmark_name) > 0
+            ), f"Telemetry data missing benchmark_name \
+                 for cloud_account_stats {account}"
+            assert (
+                len(account.posture_management_stats.benchmark_version) > 0
+            ), f"Telemetry data missing \
+                benchmark_version for cloud_account_stats {account}"
```

Comment:
any reason to split all the strings? formating here seems off, is auto-formatted like that by black?

Reply:
Yes, I used the ["black" auto reformat.](https://www.freecodecamp.org/news/auto-format-your-python-code-with-black/)

Reply:
Here's a patch that doesn't break strings like that and passes black:

```patch
From 626d0234263643bc8d9b19242593b8662cb000fb Mon Sep 17 00:00:00 2001
From: Orestis Floros <orestis.floros@elastic.co>
Date: Tue, 28 Nov 2023 13:15:13 +0100
Subject: [PATCH] fmt

---
 .../tests/test_sanity_telemetry.py            | 47 +++++--------------
 1 file changed, 12 insertions(+), 35 deletions(-)

diff --git a/tests/integration/tests/test_sanity_telemetry.py b/tests/integration/tests/test_sanity_telemetry.py
index 7da71a16..aad2c738 100644
--- a/tests/integration/tests/test_sanity_telemetry.py
+++ b/tests/integration/tests/test_sanity_telemetry.py
@@ -54,30 +54,20 @@ def test_telemetry_cloud_account_stats(cloud_security_telemetry_data):
     # account stats
     cloud_account_stats = cloud_security_telemetry_data.cloud_account_stats
     for account in cloud_account_stats:
-        assert (
-            len(account.account_id) > 0
-        ), f"Telemetry data missing account_id for \
-            cloud_account_stats {account}"
-
-        assert (
-            len(account.product) > 0
-        ), f"Telemetry data missing product for \
-             cloud_account_stats {account}"
+        assert len(account.account_id) > 0, f"Telemetry data missing account_id for cloud_account_stats {account}"
+        assert len(account.product) > 0, f"Telemetry data missing product for cloud_account_stats {account}"
 
         if not (account.product == "kspm" and "CIS Kubernetes" in account.posture_management_stats.benchmark_name):
             assert (
                 len(account.cloud_provider) > 0
-            ), f"Telemetry data missing cloud_provider for \
-                    cloud_account_stats {account}"
+            ), f"Telemetry data missing cloud_provider for cloud_account_stats {account}"
         if account.product != "vuln_mgmt":
             assert (
                 len(account.posture_management_stats.benchmark_name) > 0
-            ), f"Telemetry data missing benchmark_name \
-                 for cloud_account_stats {account}"
+            ), f"Telemetry data missing benchmark_name for cloud_account_stats {account}"
             assert (
                 len(account.posture_management_stats.benchmark_version) > 0
-            ), f"Telemetry data missing \
-                benchmark_version for cloud_account_stats {account}"
+            ), f"Telemetry data missing benchmark_version for cloud_account_stats {account}"
 
 
 @pytest.mark.sanity
@@ -91,33 +81,20 @@ def test_telemetry_installation_stats(cloud_security_telemetry_data):
     # installation stats
     installation_stats = cloud_security_telemetry_data.installation_stats
     for installation in installation_stats:
-        assert (
-            len(installation.package_policy_id) > 0
-        ), "Telemetry data missing package_policy_id in \
-                installation stats"
-        assert (
-            len(installation.feature) > 0
-        ), f"Telemetry data missing feature in installation_stats \
-                for {installation}"
-        assert (
-            installation.agent_count > 0
-        ), f"Telemetry data missing agent in installation_stats \
-                for {installation}"
+        assert len(installation.package_policy_id) > 0, "Telemetry data missing package_policy_id in installation stats"
+        assert len(installation.feature) > 0, f"Telemetry data missing feature in installation_stats for {installation}"
+        assert installation.agent_count > 0, f"Telemetry data missing agent in installation_stats for {installation}"
         assert (
             len(installation.package_version) > 0
-        ), f"Telemetry data missing package_version in installation_stats \
-                for {installation}"
+        ), f"Telemetry data missing package_version in installation_stats for {installation}"
         assert (
             len(installation.deployment_mode) > 0
-        ), f"Telemetry data missing deployment_mode in installation_stats \
-                            for {installation}"
+        ), f"Telemetry data missing deployment_mode in installation_stats for {installation}"
         if installation.feature == "cspm":
             assert isinstance(
                 installation.is_setup_automatic,
                 bool,
-            ), f"Telemetry data missing is_setup_automatic \
-                 in installation_stats for {installation}"
+            ), f"Telemetry data missing is_setup_automatic in installation_stats for {installation}"
             assert (
                 len(installation.account_type) > 0
-            ), f"Telemetry data missing account_type in installation_stats \
-                                    for {installation}"
+            ), f"Telemetry data missing account_type in installation_stats for {installation}"
-- 
2.43.0

```

Reply:
@orestisfl, This is the output from the pre-commit hook, right?

I already followed this suggestion, and when I ran the pre-commit hook again, I got the following error:

```patch
black....................................................................Failed
- hook id: black
- duration: 0.63s
- exit code: 1

--- tests/integration/tests/test_sanity_telemetry.py    2023-11-29 17:33:56.225230+00:00
+++ tests/integration/tests/test_sanity_telemetry.py    2023-11-29 17:34:03.382199+00:00
@@ -92,11 +92,11 @@
         assert (
             len(installation.deployment_mode) > 0
         ), f"Telemetry data missing deployment_mode in installation_stats for {installation}"
 
         if installation.feature == "cspm":
-            assert isinstance(installation.is_setup_automatic, bool), (
-                f"Telemetry data missing is_setup_automatic in installation_stats for {installation}"
-            )
+            assert isinstance(
+                installation.is_setup_automatic, bool
+            ), f"Telemetry data missing is_setup_automatic in installation_stats for {installation}"
             assert (
                 len(installation.account_type) > 0
             ), f"Telemetry data missing account_type in installation_stats for {installation}"
```

It forces me to break the lines this way.

Reply:
Thanks @orestisfl, The file has now been updated to adhere to the recommended pre-commit hook instructions.